### PR TITLE
Fix relay syncing

### PR DESCRIPTION
**Pull request by BlowaterNostr**
In this PR, we officially introduce [Automerge CRDT](https://automerge.org/) to implement mutable values over Nostr.

Specificly, we refactor relay configurations to use Automerge and serialize automerge doc object to hex and store it as a [kind 30078](https://github.com/nostr-protocol/nips/blob/master/78.md)

We also refactor how individual component should work nicer with MVU architecture.

### CODE REVIEW

File: UI/app.tsx [hunk 1]
```diff
@@ -55,25 +55,33 @@ export async function Start(database: DexieDatabase) {
         const dbView = await Database_Contextual_View.New(database, ctx);
         const lamport = time.fromEvents(dbView.filterEvents((_) => true));
         const app = new App(dbView, lamport, model, ctx, eventBus, pool);
-        const err = await app.initApp(ctx);
-        if (err instanceof Error) {
-            throw err;
-        }
+        const err = await app.initApp(ctx, pool);
+        // if (err instanceof Error) {
```
Comment (foodstr) [inline UI/app.tsx:59]: Why do we comment out here?

File: UI/app.tsx [hunk 2]
```diff
@@ -166,18 +192,23 @@ export class App {
...
-                this.relayConfig.pool.addRelayURLs(defaultRelays);
+            {
+                // relay config synchronization
+                for (const e of events) {
+                    const _relayConfig = await RelayConfig.FromNostrEvent(e, this.myAccountContext);
+                    if (_relayConfig instanceof Error) {
+                        console.log(_relayConfig.message);
+                        continue;
+                    }
+                    this.relayConfig.merge(_relayConfig.save());
```
Comment (foodstr) [inline UI/app.tsx:203]: nice

File: UI/app.tsx [hunk 3]
```diff
@@ -150,10 +158,28 @@ export class App {
     ) {
         this.eventSyncer = new EventSyncer(relayPool, this.database);
         this.allUsersInfo = new AllUsersInformation(myAccountContext);
-        this.relayConfig = new RelayConfig(relayPool, this.myAccountContext);
+        this.relayConfig = new RelayConfig();
+        if (this.relayConfig.getRelayURLs().size == 0) {
```
[the remark this answers is not in the document]
Comment (BlowaterNostr) [reply UI/app.tsx:162]: Yes. The `RelayConfig` contrusctor used to load from localstorage when I first wrote it. But later I feel I need to implement storage IO separately. Haven't done that yet.

File: UI/app.tsx (hunk 1 above)
Comment (BlowaterNostr) [reply UI/app.tsx:59]: for debug. will change.

**Pull request merged**